server/src/main/java/org/elasticsearch/cluster/routing/ExpectedShardSizeEstimator.java

```diff
@@ -43,6 +43,8 @@ public static long getExpectedShardSize(
         Metadata metadata,
         RoutingTable routingTable
     ) {
+        assert defaultValue == 0L || defaultValue == ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE
```

Comment:
In DiskThresholdDecider.java, there's a place
```
totalSize += getExpectedShardSize(
                    routing,
                    Math.max(routing.getExpectedShardSize(), 0L),
                    clusterInfo,
                    null,
                    metadata,
                    routingTable
                );
```

This seems to set the default value to other values?